apis/discoveryengine/v1alpha1/engine_reference.go

```diff
@@ -43,16 +43,22 @@ type DiscoveryEngineEngineRef struct {
 	Namespace string `json:"namespace,omitempty"`
 }
 
+// DiscoveryEngineEngineID is the resolved identifier for a DiscoveryEngineEngine
+type DiscoveryEngineEngineID struct {
+	*CollectionLink
+	Engine string
+}
+
 // NormalizedExternal provision the "External" value for other resource that depends on DiscoveryEngineEngine.
 // If the "External" is given in the other resource's spec.DiscoveryEngineEngineRef, the given value will be used.
 // Otherwise, the "Name" and "Namespace" will be used to query the actual DiscoveryEngineEngine object from the cluster.
-func (r *DiscoveryEngineEngineRef) NormalizedExternal(ctx context.Context, reader client.Reader, otherNamespace string) (string, error) {
+func (r *DiscoveryEngineEngineRef) NormalizedExternal(ctx context.Context, reader client.Reader, defaultNamespace string) (string, error) {
```

Comment:
I'd prefer "other" over "default" because this is the referrer's namespace, which overrides the "default". 
e.g.
```yaml
kind: OtherResource
metadata:
   name: other
   namespace: otherNamespace
spec:
    discoveryEnginRef:
         name:  this 
--- 
kind: DiscoveryEngineEngine
metadata:
   name: this
```
The `DiscoveryEngineEngine`  will use namespace "otherNamespace"

Reply:
OK, I'll update

Reply:
My thought was that this is actually the default value for the Namespace of the Ref.  That's the only namespace the _ref_ has.

Reply:
Just pushed changed with otherNamespace.  We probably need to cleanup the NormalizedExternal contract anyway so maybe we document it there